atom/browser/api/lib/app.coffee

```diff
@@ -32,8 +32,10 @@ app.setAppPath = (path) ->
 app.getAppPath = ->
   appPath
 
-# Be compatible with old API.
-app.once 'ready', -> @emit 'finish-launching'
+app.once 'ready', ->
+  app.defaultSession.__proto__ = EventEmitter.prototype
```

Comment:
We should wrap the `session` like what we did for `webContents`, you can start from `_setWrapWebContents`.